### PR TITLE
rustdesk-flutter: init at unstable-2024-02-03

### PR DESCRIPTION
**Pull request by dasJ**
## Description of changes

This is a second package to ensure people can still use the old one which is built from a stable tag. The old 1.2.3 version uses a flutter that is too old which is why it was dropped from nixpkgs.

If anyone wants to be added as a maintainer as well (@ocfox, @Leixb, @philipwilk), feel free to add a comment ;)

Supersedes #281906 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by philipwilk** [comment]
Seems to build and run for me, but I unfortunately I cant test it right at this moment as xdg-desktop-portal-wlr doesn't support org.freedesktop.portal.RemoteDesktop yet, will see if I can spin up gnome again tomorrow to give it a shot, but I have been able to test that i can connect to my phone and that seems to be working all fine.

Don't want to delay this pull as it's really needed just to finally have hydra caching of this package (rip laptops), but I think it would be worth looking into getting the other rustdesk crate features working in our builds in a future PR.

**Comment by dasJ** [comment]
I looked at them and most of them did android things (like pulling in the ndk) or activated alternative audio resamplers. If there are specific features you think are worth exploring, please don't hesitate to mention them here but from what I can tell, the features I enabled are all that make sense.

**Comment by 0x4A6F** [comment]
Result of `nixpkgs-review pr 286129` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>rustdesk-flutter</li>
    <li>rustdesk-flutter.debug</li>
    <li>rustdesk-flutter.pubcache</li>
  </ul>
</details>

**Review by 0x4A6F — APPROVED** [review]

**Comment by philipwilk** [comment]
> Seems to build and run for me, but I unfortunately I cant test it right at this moment as xdg-desktop-portal-wlr doesn't support org.freedesktop.portal.RemoteDesktop yet, will see if I can spin up gnome again tomorrow to give it a shot, but I have been able to test that i can connect to my phone and that seems to be working all fine.
> 
> Don't want to delay this pull as it's really needed just to finally have hydra caching of this package (rip laptops), but I think it would be worth looking into getting the other rustdesk crate features working in our builds in a future PR.

Tested connecting to a gnome 45 desktop on wayland from my phone and it works perfectly (unlike the old sciter version), great!

File: pkgs/by-name/ru/rustdesk-flutter/package.nix [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+        exec = "${meta.mainProgram} %u";
+      };
+    })
+    (makeDesktopItem {
+      name = "rustdesk-link";
+      desktopName = "RustDeskURL Scheme Handler";
+      noDisplay = true;
+      mimeTypes = [ "x-scheme-handler/rustdesk" ];
+      tryExec = "rustdesk";
+      exec = "${meta.mainProgram} %u";
```
Comment (SuperSandro2000) [inline pkgs/by-name/ru/rustdesk-flutter/package.nix:198]: A change in meta should never trigger a cache miss.

File: pkgs/by-name/ru/rustdesk-flutter/package.nix [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+    };
+
+    cargoHash = "sha256-dDyiptG9TKes+fXx2atwx697SWH7Rltx6xVubtTn7FM=";
+    cargoBuildFlags = [ "--package" "flutter_rust_bridge_codegen" ];
+    doCheck = false;
+  };
+
+in flutter313.buildFlutterApplication rec {
+  pname = "rustdesk";
+  version = "unstable-2024-02-03";
```
Comment (SuperSandro2000) [inline pkgs/by-name/ru/rustdesk-flutter/package.nix:41]:
```suggestion
  version = "1.2.3-unstable-2024-02-03";
```
According to the contributing guide we should do this

**Pull request merged**